dataland-backend/src/main/kotlin/org/dataland/datalandbackend/controller/DataController.kt

```diff
@@ -1,43 +1,54 @@
 package org.dataland.datalandbackend.controller
 
+import com.fasterxml.jackson.databind.ObjectMapper
 import org.dataland.datalandbackend.api.DataAPI
 import org.dataland.datalandbackend.interfaces.DataStoreInterface
-import org.dataland.datalandbackend.model.DataSet
+import org.dataland.datalandbackend.model.DataIdentifier
 import org.dataland.datalandbackend.model.DataSetMetaInformation
-import org.dataland.skyminderClient.interfaces.DataConnectorInterface
-import org.dataland.skyminderClient.model.ContactInformation
+import org.dataland.datalandbackend.model.StorableDataSet
 import org.springframework.beans.factory.annotation.Autowired
-import org.springframework.beans.factory.annotation.Qualifier
 import org.springframework.http.ResponseEntity
 import org.springframework.web.bind.annotation.RestController
 
 /**
  * Implementation of the API for data exchange
  * @param dataStore implementation of the DataStoreInterface that defines how uploaded data is to be stored
- * @param dataConnector implementation of the DataConnectorInterface that defines how to connect to the data
- * source (e.g. skyminder)
  */
 
 @RestController
-class DataController(
-    @Autowired @Qualifier("DefaultStore") var dataStore: DataStoreInterface,
-    @Autowired var dataConnector: DataConnectorInterface
-) : DataAPI {
+abstract class DataController<T>(
+    @Autowired var dataStore: DataStoreInterface,
+    @Autowired var objectMapper: ObjectMapper,
+    val clazz: Class<T>
+) : DataAPI<T> {
```

Comment:
Bitte `@RestController` und `@Autowired` hier weg machen: Tatsächlich wird diese Klasse ja gar nicht von Spring gebaut, sondern in der jeweiligen Implementierung per `super()`. Deshalb können die Dinge hier auch nicht autowired werden - und Spring ignoriert die Annotation einfach.

Reply:
Fixed an dieser Stelle und auch nur an dieser Stelle, da ja einzige abstrakte Klasse.